### ISSUE TITLE
Selectizes destroy when page animation starts

### ISSUE THREAD

**Issue by plashenkov** [rhyek/vue2-selectize#7]
Hi!

Firstly, thanks for the superb component!

I use page transitions and have noticed that component is destroyed _before_ animation begins. Thus, standard selects become visible and layout is distorted at the time of the animation.

See the screencast:
http://take.ms/70FsB

Currently I use following workaround: simply apply styles to standard select (`opacity: 0; height: ...px;`). This helps, but selectize components disappear abruptly, this is drawback.

Is there any method to achieve selectizes destroying on transition end instead of begin?

**Comment by rhyek** [comment]
Hm, this is interesting. I honestly don't have any experience working with transitions. I just checked and I am destroying the `selectize` instance using Vue's `destroyed` lifecycle hook. That is pretty much the last chance you get to do this sort of work.

I suspect this will have more to do with how you configure the transition. Would you mind looking into it and letting me know if you learn anything? My time is sort of limited, currently.

**Comment by autumnwoodberry** [comment]
https://forum.vuejs.org/t/vue-destroys-components-on-a-page-when-they-are-still-needed/20560/5?u=woodberry

**Comment by plashenkov** [comment]
https://github.com/vuejs/vue/issues/6983

**Comment by rhyek** [comment]
@plashenkov, thanks. I just subscribed to that issue. I'll leave this one open for now, but it seems clear to me it is a framework issue.

I'll follow the conversation for a few days, but if nothing comes of it the best I could do is to implement an option to not destroy the `selectize` instance on `destroyed`.

**Comment by rhyek** [comment]
Yeah, an option not automatically destroy the instance and then you could do something like `this.$refs.someSelectize.$el.selectize.destroy()` in your parent component using the `transitionend` event mentioned here.

**Comment by rhyek** [comment]
Actually, if you want to provide a PR for that I'd be happy to accept and publish.

**Comment by plashenkov** [comment]
@rhyek Are you absolutely sure we really need to `destroy()` selectizes? Maybe, when component is removed from the DOM by Vue, this is all we need?

**Comment by rhyek** [comment]
The new nodes that are generated by selectize are not managed by Vue.

On Nov 1, 2017 1:01 PM, "Yury Plashenkov" <notifications@github.com> wrote:

@rhyek <https://github.com/rhyek> Are you absolutely sure we really need to
destroy() selectizes? Maybe, when component is removed from the DOM by Vue,
this is all we need?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/rhyek/vue2-selectize/issues/7#issuecomment-341206275>,
or mute the thread

**Comment by plashenkov** [comment]
Well... for option `dropdownParent: 'body'`, that's true. Otherwise, it is remove completely from the DOM as I can see.

**Comment by rhyek** [comment]
Maybe it is removed in your case because you are destroying the entire
page.
